File: EnterpriseLeaveManagement.Domain/Entities/Department.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using EnterpriseLeaveManagement.Domain.Common;

namespace EnterpriseLeaveManagement.Domain.Entities;

public class Department : BaseEntity
{
    public string Name { get; set; } = string.Empty;

    public string Code { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;
    public bool IsActive { get; set; } = true;

    //Foreign Keys
    public Guid? ManagerId { get; set; }

    //Navigation properties
    public ICollection<Employee> Employees { get; set; } = new List<Employee>();

}

File: EnterpriseLeaveManagement.UI/src/types/department.ts
export interface Department {
	id: string;
	name: string;
	code: string;
	description: string;
	isActive: boolean;
}

export interface DepartmentApiDto {
	id: string;
	name: string;
	code: string;
	description: string;
	isActive: boolean;
}

export interface CreateDepartmentRequest {
	name: string;
	code: string;
	description: string;
	managerId?: string | null;
}

export interface UpdateDepartmentRequest
	extends CreateDepartmentRequest {
	id: string;
}
